### PR TITLE
Updating docs - Added an example to sampling and added sampling to the docs

### PR DESCRIPTION
**Pull request by springcoil**
These can probably be moved around a bit. 

https://github.com/pymc-devs/pymc3/issues/1968

### CODE REVIEW

File: docs/source/api/data.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+*****
```
Comment (aseyboldt) [inline docs/source/api/data.rst:1]: You need to add new files to the toctree in `source/api.rst`, or they won't show up in the documentation.

File: docs/source/api/sampling.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+********
```
Comment (aseyboldt) [inline docs/source/api/sampling.rst]: This is already in `inference.rst`.

File: pymc3/sampling.py [hunk 3]
```diff
@@ -149,6 +149,30 @@ def sample(draws, step=None, init='ADVI', n_init=200000, start=None,
...
+        ... np.random.seed(20090425)
+        ... n = 20
+        ... X = np.sort(3*np.random.rand(n))
+
+    Now let's set up a model and sample from it.
+
+    .. code:: ipython
+        >>> with pm.Model() as model:
+        ...  b0 = pm.Normal('b0', mu=0, sd=100)
+        ...  yest = b0 * X
```
Comment (aseyboldt) [inline pymc3/sampling.py]: This doesn't do anything. Did you forget an observed var here?

File: pymc3/sampling.py [hunk 4]
```diff
@@ -149,6 +149,30 @@ def sample(draws, step=None, init='ADVI', n_init=200000, start=None,
...
+        ...
+        >>> with model:
+        ...      trc = pm.sample(100)
```
Comment (aseyboldt) [inline pymc3/sampling.py]: I'd prefer `trace = pm.sample(2000, tune=1000, njobs=4)`

File: pymc3/sampling.py [hunk 5]
```diff
@@ -149,6 +149,30 @@ def sample(draws, step=None, init='ADVI', n_init=200000, start=None,
     -------
     trace : pymc3.backends.base.MultiTrace
         A `MultiTrace` object that contains the samples.
+
+        Examples
```
Comment (aseyboldt) [inline pymc3/sampling.py]: Good idea, that was definitely missing.
nit: bad indent

**Comment by springcoil** [comment]
Updated this - thanks.

**Pull request merged**